docs/backend.md

```diff
@@ -2,40 +2,271 @@
+{
+    return array_filter((array)$this, function ($value) {
+        return $value !== null;
+    });
+}
+```
+The implementation has to **remove all null variables**.
+
+**[Converter](../api/src/JMP/Utils/Converter.php):**
+This util is used to convert a model object or  a list of model objects properly to an associative array.
+Use it in the controller as shown in this examples:
+```php
+return $response->withJson(Converter::convert($modelObject));
+```
+or
+```php
+return $response->withJson(Converter::convertArray($list));
+```
+
+## Type declarations & phpdoc
+Everywhere it is possible we use object oriented php. So every method signature have to use the [php type declarations](https://secure.php.net/manual/en/functions.arguments.php#functions.arguments.type-declaration). If you try to call a method and pass arguments of another type as declared, then php will throw an [TypeError](https://secure.php.net/manual/en/class.typeerror.php).
+
+Also, every method has to be documented with phpdoc. A short summary or description of the method and a documented signature is sufficient.
+
+**Example (from [UserService.php](../api/src/JMP/Services/UserService.php)):**
+```php
+/**
+ * Select a user by its id
+ * @param int $userId
+ * @return Optional containing a User on succeed
+ */
+public function getUserByUserId(int $userId): Optional
+```
+
+## SQL
+The services handle the business logic as well as the whole database communication. At the moment there isn't a separated Persistence Layer with e.g. Data Access Objects (DAOs).
+
+That the services doesn't become a mess, all SQL have to be coded with the following rules:
+* Use the heredoc syntax:
+```php
+$sql = <<<SQL
+# SQL
+SQL;
+```
+* Write every sql keyword upper-case
+* Use the **`AS`** keyword to change column names from the sql underscore style to the camel case style. E.g. `SELECT user_id as userId ...`
+* Use the following for optional parameters: `WHERE (:optionalId IS NULL OR id = :optionalId)`. Examples in [EventService.php](../api/src/JMP/Services/EventService.php)
+* A specific service don't query tables to which the service doesn't belong unless it's a join
+
+
+# Developing a new route
+This guide will take you through the most important steps to develop a new route.
+Make sure you've read [Code Style](#code-style) before reading the following guide.
+
+## Api specification
+First of all the new route has to be documented in the [api-specification](api-v1.md).
 
 ## Service & Model
+Routes return JSON mostly containing one or a list of model objects. So you have to [create a new model](#create-a-new-model) if the required doesn't exist yet. The model is just a raw data object with the functionality to convert properly to an array.
+
+Then you need a service to communicate with the database and to handle the business logic.
+[Create a new service class](#create-a-new-service-class) if no appropriate exists, otherwise add the required method/s to the existing service class.
```

Comment:
```suggestion
[Create a new service class](#create-a-new-service-class) if no appropriate exists yet, otherwise add the required method/s to the existing service class.
```